### ISSUE TITLE
ampq extension

### ISSUE THREAD

**Issue by dgoosens** [jakzal/phpqa#344]
h

this is not a bug.... but figure it might be useful for others... not sure what the best place would be

So I needed the PHP AMPQ extension...  
This can not be installed with the classic `RUN docker-php-ext-install amqp` command as the extension is not available yet (there always is some time after the release of a PHP version before that extension becomes available)

This is how you can proceed to install it from source:

As explained in the documentation, you should create a local `Dockerfile` and extend the `jakzal/phpqa` image  
(I will be using the Debian version here)

```
FROM jakzal/phpqa:debian AS ext-amqp
ENV EXT_AMQP_VERSION=master

RUN docker-php-source extract
RUN apt-get update
RUN apt-get install -qq -y build-essential autoconf librabbitmq-dev
RUN git clone --branch $EXT_AMQP_VERSION --depth 1 https://github.com/php-amqp/php-amqp.git /usr/src/php/ext/amqp
RUN cd /usr/src/php/ext/amqp && git submodule update --init
RUN docker-php-ext-install amqp

FROM jakzal/phpqa:debian

COPY --from=ext-amqp /usr/local/etc/php/conf.d/docker-php-ext-amqp.ini /usr/local/etc/php/conf.d/docker-php-ext-amqp.ini
COPY --from=ext-amqp /usr/local/lib/php/extensions/no-debug-non-zts-20210902/amqp.so /usr/local/lib/php/extensions/no-debug-non-zts-20210902/amqp.so

RUN apt-get update
RUN apt-get install -qq -y librabbitmq-dev
```

(I'm sorry, but I don't remember where I read about this procedure)

**Comment by jakzal** [comment]
Thanks. 

The README file has become quite big over the years. I was thinking to create a cookbook where recipies like this one could live (i.e. `docs/cookbook/`).

**Comment by dgoosens** [comment]
Good idea...  
Would you like me to create an MR with that ?

**Comment by jakzal** [comment]
Fixed by #345.
